cfme/tests/infrastructure/test_provider_discovery.py

```diff
@@ -103,14 +105,18 @@ def test_discover_infra(providers_for_discover, start_ip, max_range):
 
     discover(rhevm, virtualcenter, scvmm, False, start_ip, max_range)
 
-    @pytest.wait_for(num_sec=count_timeout(start_ip, max_range), delay=5)
-    def _wait_for_all_providers():
-        for provider in providers_for_discover:
-            if provider.key not in pytest.store.current_appliance.managed_providers:
+    try:
+        @pytest.wait_for(num_sec=count_timeout(start_ip, max_range), delay=5)
```

Comment:
@jistefan `utils.wait.wait_for_decorator` ... wait, we don't wrap that one.

```python
# utils/wait.py
# modify
from wait_for import wait_for as wait_for_mod, wait_for_decorator as wait_for_decorator_mod

# add at the end
wait_for_decorator = partial(wait_for_decorator_mod, logger=logger)
```

Then in your file:

```python
from utils.wait import wait_for_decorator

@wait_for_decorator(num_sec=count_timeout(start_ip, max_range), delay=5)
def foo():
    bar()
```